### PR TITLE
[Keras 3 OpenVINO Backend]: Improve support for `unstack` operations

### PR DESCRIPTION
**Pull request by daehyun99**
## Details :

Hi, there.

Implements `unstack` operation for the OpenVINO backend.

I will implement it while addressing this issue.

Thanks.

### milestone 1
- [x] Test: Activate Test code (TDD)
- [x] Feat: Implement OpenVINO's backend `unstack`

## Closes issue :

- Closes: https://github.com/openvinotoolkit/openvino/issues/34619

### CODE REVIEW

**Comment by daehyun99** [comment]
@hertschuh  @goyaladitya05 

Hi, devs

I have a question.

It seems to me that the `unstack` operation exists in two different places within the OpenVINO backend.

https://github.com/keras-team/keras/blob/dc3bdb140db317677faf2c96e3a5915582fff69d/keras/src/backend/openvino/core.py#L1531-L1534

https://github.com/keras-team/keras/blob/dc3bdb140db317677faf2c96e3a5915582fff69d/keras/src/backend/openvino/rnn.py#L295-L298

Would it be appropriate to remove the `unstack` in `rnn.py` and instead implement the `unstack` in `core.py`?

Thanks.

**Comment by goyaladitya05** [comment]
> Would it be appropriate to remove the `unstack` in `rnn.py` and instead implement the `unstack` in `core.py`?

I compared with other backends, the code is also duplicated for JAX and Numpy. Whereas Torch and Tensorflow has it in `core.py` only. I think it should be in `core.py` only. 

Also one more thing, we do not need to create this issue https://github.com/openvinotoolkit/openvino/issues/34697 in OpenVINO, since that improvement is solely a part of Keras. Thanks!

**Comment by daehyun99** [comment]
@goyaladitya05 

> I compared with other backends, the code is also duplicated for JAX and Numpy. Whereas Torch and Tensorflow has it in core.py only. I think it should be in core.py only.

Okay. I understand.

> Also one more thing, we do not need to create this issue https://github.com/openvinotoolkit/openvino/issues/34697 in OpenVINO, since that improvement is solely a part of Keras. Thanks!

Oh, that makes sense.
I think I missed that.
I'll close the issue in OpenVINO and create a new one in Keras.
Thanks for the quick response.

Thanks.

**Comment by goyaladitya05** [comment]
Hey @daehyun99, 
Please don't remove `unstack` from `rnn.py` in this PR. I'm implementing the functions for rnn.py in #22436 so I'm removing the duplicate entry of unstack there itself. This should help us avoid any merge conflicts when both the PRs get merged. Thanks!

**Comment by daehyun99** [comment]
@goyaladitya05 
Yes, I understand.
i will do not change `rnn.py` file.

Thanks.

**Comment by daehyun99** [comment]
Milestone 1 is Done.

- This test was passed.
```sh
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsDynamicShapeTest::test_unstack
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsStaticShapeTest::test_unstack
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsBehaviorTests::test_unstack_unknown_axis_num
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsCorrectnessTest::test_unstack
```

Thanks.

File: keras/src/backend/openvino/core.py [hunk 1]
```diff
@@ -1529,9 +1529,23 @@ def stop_gradient(variable):
...
+    x_ov = get_ov_output(x)
+    axis_ov = get_ov_output(axis)
+
+    if num is None:
+        axis_val = axis if isinstance(axis, int) else axis_ov.get_data().item()
+        shape = x_ov.get_partial_shape()
+        num = shape[axis_val].get_length()
+    elif not isinstance(num, int):
+        num_ov = get_ov_output(num)
+        num = num_ov.get_data().item()
```
Comment (hertschuh) [inline keras/src/backend/openvino/core.py]: This can be simplified because `num` and `axis` are ints, not tensors.

Comment (daehyun99) [reply keras/src/backend/openvino/core.py]: @hertschuh 
I apply your review.
Originally, I intended to address #22423 in this PR as well, but I will create a separate PR for it.

Review please.

Thanks.

[the remark this answers is not in the document]
Comment (daehyun99) [reply keras/src/backend/openvino/core.py]: The tests below passed, so I'm not sure if this request is actually valid.

```
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsDynamicShapeTest::test_unstack
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsStaticShapeTest::test_unstack
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsBehaviorTests::test_unstack_unknown_axis_num
KERAS_BACKEND=openvino pytest keras/src/ops/core_test.py::CoreOpsCorrectnessTest::test_unstack
```
